README.Rmd

```diff
@@ -18,51 +18,137 @@ knitr::opts_chunk$set(
+You can install the development version of `{simulist}` from
 [GitHub](https://github.com/) with:
 
 ``` r
-# install.packages("pak")
-pak::pak("{{ gh_repo }}")
+# check whether {pak} is installed
+if(!require("pak")) install.packages("pak")
+pak::pak("epiverse-trace/simulist")
 ```
 
-## Example
+## Quick start
 
-These examples illustrate some of the current functionalities
+```{r load-simulist}
+library(simulist)
+library(epiparameter)
+```
+
+The line list simulation requires a serial interval, onset-to-hospitalisation delay, and onset-to-death delay. We can load these load these from the library of epidemiological parameters in the `{epiparameter}` R package, or if these are not available, such as the serial interval for COVID-19 we can create them ourselves.
 
+```{r create-epidists}
+# create COVID-19 serial interval
+serial_interval <- epiparameter::epidist(
+  disease = "COVID-19",
+  epi_dist = "serial interval",
+  prob_distribution = "gamma",
+  prob_distribution_params = c(shape = 1, scale = 1)
+)
 
-## Development
+# get onset to hospital admission from {epiparameter} database
+onset_to_hosp <- epiparameter::epidist_db(
+  disease = "COVID-19",
+  epi_dist = "onset to hospitalisation",
+  single_epidist = TRUE
+)
+
+# get onset to death from {epiparameter} database
+onset_to_death <- epiparameter::epidist_db(
+  disease = "COVID-19",
+  epi_dist = "onset to death",
+  single_epidist = TRUE
+)
+```
 
-### Lifecycle
+To simulate a line list for COVID-19 with an assumed reproduction number (`R`) of 1.1 we use the `sim_linelist()` function.
 
-This package is currently a *concept*, as defined by the [RECON software
-lifecycle](https://www.reconverse.org/lifecycle.html). This means that essential
-features and mechanisms are still being developed, and the package is not ready
-for use outside of the development team.
+```{r sim-linelist}
+linelist <- sim_linelist(
+  R = 1.1,
+  serial_interval = serial_interval,
+  onset_to_hosp = onset_to_hosp,
+  onset_to_death = onset_to_death
+)
+head(linelist)
+```
 
+In this example, the line list is simulated using the default values (see `?sim_linelist`). The default hospitalisation rate is 0.2 (or 20% of individual infected become hospitalised) and the start date of the outbreak is 1st January 2023. To modify either of these to make them more realistic we can specify them.
 
-### Contributions
+```{r sim-linelist-diff-args}
+linelist <- sim_linelist(
+  R = 1.1,
+  serial_interval = serial_interval,
+  onset_to_hosp = onset_to_hosp,
+  onset_to_death = onset_to_death,
+  hosp_rate = 0.01,
+  outbreak_start_date = as.Date("2019-12-01")
+)
+head(linelist)
+```
 
-Contributions are welcome via [pull requests](https://github.com/{{ gh_repo }}/pulls).
+To simulate a table of contacts we can reuse the serial interval from the example above, and we additionally need a contact distribution. This distribution represents the variability in number of contacts that each person in the population has.
```

Comment:
```suggestion
To simulate a table of contacts of cases (i.e. to a reflect contact tracing dataset) we can reuse the serial interval from the example above. We additionally need a contact distribution, which represents the probability that each person in the population will be a given number of contacts on a given day.
```

Reply:
Edited definition – correct?

Reply:
I've updated the definition in 6961c68c87bde43edc0a9f160647fc061a6da27e and 907457c55da9029edfea71188df6c774fa3d528e. I think we need to think a bit more about the definition and implementation of this before the release, or at least for the next release. Discussion is taking place in issue #35.